File: Max Number of K-Sum Pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& num, int k) {
      sort(num.begin(), num.end());
      int c = 0;
      int i=0,j=num.size()-1;
      while(i < j){
        if(num[i] + num[j] == k){
          c++; 
          i++; 
          j--;
        } 
		else if(num[i] + num[j] > k)
            j--;
        else
            i++;
    }
      return c;
    }
};

File: leet-code-solutions/max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        left=0
        right=len(nums)-1
        count=0

        nums.sort()

        while left < right:
            if nums[left] + nums[right] == k:
                right-=1
                left+=1
                count+=1
            elif nums[left] + nums[right] < k:
                left+=1
            else:
                right-=1
        
        return count